kernel/src/hil/analog_comparator.rs

```diff
@@ -0,0 +1,30 @@
+//! Interface for direct control of the analog comparators.
+
+// Author: Danilo Verhaert <verhaert@cs.stanford.edu>
+// Last modified August 7th, 2018
+
+use returncode::ReturnCode;
+
+pub trait AnalogComparator {
+    /// The chip-dependent type of an analog comparator channel.
+    type Channel;
+
+    /// Do a single comparison of two inputs, depending on the AC chosen. Output
+    /// will be True (1) when one is higher than the other, and False (0)
+    /// otherwise.  Specifically, the output is True when Vp > Vn (Vin positive
+    /// > Vin negative), and False if Vp < Vn.
+    fn comparison(&self, channel: &Self::Channel) -> bool;
+
+    /// Start interrupt-based comparison for the chosen channel (e.g. channel 1
+    /// for AC1). This will make it listen and send an interrupt as soon as
+    /// Vp > Vn.
+    fn start_comparing(&self, channel: &Self::Channel) -> ReturnCode;
+
+    /// Stop interrupt-based comparison for the chosen channel.
+    fn stop_comparing(&self, channel: &Self::Channel) -> ReturnCode;
+}
+
+pub trait Client {
+    /// Called when an interrupt occurs.
+    fn fired(&self, usize);
```

Comment:
More documentation needed, in particular how this unnamed usize maps to the `Channel` above.

Reply:
It doesn't. Currently, by checking the interrupt status register the used channel is determined, which is done by some simple `if` statements: I don't believe it's worth it to add to the complexity by adding channels here: I've given it a shot, but it got too complex. I noticed the ADC also immediately convert the Channel to a `usize` as well in the chip-specific code/capsule; it only uses the enum to make sure a valid channel is addressed by checking ` self.channels.len()`

If you do believe there is a better way by keeping Channel in and keeping things clear let me know: in that case, we should change it in the ADC as well (basically, I've taken the same approach as there).

Reply:
I don't have an issue with using the `usize` here, just that the comment needs to explain what it is.

Reply:
Still need comment here.

Reply:
Oops, it appears I added in the capsule not the hil. Done now!